### ISSUE TITLE
PostMessage doesn't return error if error occurs

### ISSUE THREAD

**Issue by usk81** [slack-go/slack#294]
I requested to post message to slack with a incorrect channel.
expected error (`channel_not_found`) but returned nil.

code:
```go
func PostToSlack(channel string, msg string, params ...slack.PostMessageParameters) (err error) {
	token := (MY SLACK TOKEN)

	api := slack.New(token)
        api.SetDebug(true) // for debug
	p := slack.PostMessageParameters{}
	if len(params) > 0 {
		p = params[0]
	}
	// _, _, err = api.PostMessage(channel, msg, p)
	a, b, err = api.PostMessage(channel, msg, p) // for debug
        spew.Dump(a, b, err) // dump message for debug -- github.com/davecgh/go-spew/spew
	if err != nil {
		return err
	}
	return
}
```

test:
```go
func TestPostSlackWithInvalidChannel(t *testing.T) {
	err := PostToSlack("#foobar", "test")
	assert.EqualError(t, err, "channel_not_found")
}
```

test result:
```
=== RUN   TestPostSlackWithInvalidChannel
nlopes/slack2018/04/05 16:35:28 <autogenerated>:1: parseResponseBody: {"ok":false,"error":"channel_not_found"}
(string) ""
(string) ""
(interface {}) <nil>
--- FAIL: TestPostSlackWithInvalidChannel (0.22s)
	notify_test.go:37: 
			Error Trace:	notify_test.go:37
			Error:      	An error is expected but got nil.
			Test:       	TestPostSlackWithInvalidChannel
FAIL
```

**Comment by james-lawrence** [comment]
closing in favor of #307, which includes this fix and some error handling cleanup and a test.

**Comment by james-lawrence** [comment]
resolved #307
